ui/app/components/auth/form/okta.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import AuthBase from './base';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+import uuid from 'core/utils/uuid';
+
+import type AuthService from 'vault/vault/services/auth';
+
+/**
+ * @module Auth::Form::Okta
+ * see Auth::Base
+ * */
+
+export default class AuthFormOkta extends AuthBase {
+  @service declare readonly auth: AuthService;
+
+  @tracked challengeAnswer = '';
+  @tracked oktaVerifyError = '';
+  @tracked showNumberChallenge = false;
+
+  loginFields = [{ name: 'username' }, { name: 'password' }];
+
+  login = task(
+    waitFor(async (data) => {
+      // wait for 1s to wait to see if there is a login error before polling
+      await timeout(1000);
+
+      data.nonce = uuid();
+      this.pollForOktaNumberChallenge.perform(data.nonce, data.path);
```

Comment:
It would be nice if you could set up polling like you have done here and then call `super.login()` since that logic looks copied. Not sure how that would work since it's wrapped in the task and waitFor decorators though.

Reply:
yep, this is sort of the plan once all the auth method components are built. I've gone back and forth on the best way to simplify the auth logic 😵. I think I've landed on calling `this.auth.authenticate` and manually managing the login logic in each component. Then once all the components are built and tests written, it will be easier to see the best way to refactor. The end goal is to cleanup and remove the abstracted logic from the cluster adapter - which will change how the auth.authenticate method is called.